java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -98,5 +104,40 @@ protected ArrowBuf releaseBuffer(ArrowBuf buffer) {
   protected static int getValidityBufferSizeFromCount(final int valueCount) {
     return (int) Math.ceil(valueCount / 8.0);
   }
+
+  /* round up to the next multiple of 8 */
+  protected static long roundUp8(long size) {
+    return ((size + 7) / 8) * 8;
+  }
+
+  /*
+   * Allocate a single buffer and slice into multiple.
+   */
+  protected static ArrowBuf[] allocAndSlice(BufferAllocator allocator, int[] lengths) {
```

Comment:
This seems more generic than necessary. Why not make it size aware and do the allocation based on a provided value count? Also, if int[] is always length two in usage, let's make it focus on that case. For example, we shouldn't be allocating the variable data chunk in the same vector since it will have a different lifecycle.

Reply:
I want to use this for both fixed-len (data+validity) and var-len vectors (offsets+validity). 

- For the fixed-len vectors, the data buffers and validity buffers are sized for the same number-of-elements. 
- For the var-len vectors, the offset buffer needs to be sized for one extra element than the bitmap.